packages/frontend/web-ui/src/auth/pages/Login.spec.tsx

```diff
@@ -176,41 +176,40 @@ describe(Login.name, () => {
   });
 
   describe('when called, and exist a backend error and error is displayed', () => {
-    let pErrorBackend;
+    let pErrorBackend: string | null;
     const backendErrorFixture = 'backend error';
 
     beforeAll(() => {
-      useLoginForm.mockReturnValue({
+      (useLoginForm as jest.Mock<typeof useLoginForm>).mockReturnValue({
         backendError: backendErrorFixture,
         formFields: formFieldsFixture,
-        formStatus: STATUS_LOG_BACKEND_KO,
+        formStatus: LoginStatus.backendKO,
         formValidation: {},
         notifyFormFieldsFilled: notifyFormFieldsFilledMock,
         setFormField: setFormFieldMock,
       });
 
-      useShowPassword.mockReturnValue({
+      (useShowPassword as jest.Mock<typeof useShowPassword>).mockReturnValue({
         showPassword: false,
         handleClickShowPassword: handleClickShowPasswordMock,
         handleMouseDownPassword: handleMouseDownPasswordMock,
       });
 
-      useSelector.mockImplementation(() => ({
-        token: tokenFixture,
-        errorMessage: errorMessageFixture,
-      }));
+      (
+        useAppSelector as unknown as jest.Mock<typeof useAppSelector>
+      ).mockReturnValueOnce(tokenFixture);
 
-      render(
+      const renderResult: RenderResult = render(
         <MemoryRouter>
           <Login />
         </MemoryRouter>,
       );
 
-      const formErrorMessageAlert = screen.getByLabelText('form-login-error');
-      const formErrorMessageAlertMessage =
-        formErrorMessageAlert.querySelector('.MuiAlert-message');
+      const formErrorMessageAlertMessage: Text | null =
+        (renderResult.container.querySelector('.MuiAlert-message')
+          ?.childNodes[1] as Text) ?? null;
 
-      pErrorBackend = formErrorMessageAlertMessage.childNodes[1].data;
+      pErrorBackend = formErrorMessageAlertMessage.textContent;
     });
 
     afterAll(() => {
```

Comment:
`useLoginForm`, `useShowPassword` must be reset here if we use `mockReturnValue`.

Reply:
Done!